### PR TITLE
Make upgradeTo and upgradeToAndCall public

### LINKED ISSUE #3923: Make upgradeTo functions public
The `upgradeTo` and `upgradeToAndCall` functions are `external virtual`. As a result even though they can be overriden they can't be invoked via `super`, which is the correct way to override a function in this library.

From https://github.com/OpenZeppelin/openzeppelin-contracts/issues/3870.

### PR DESCRIPTION
**Pull request by m-waqas88**
Fixes #3923 

Two functions upgradeTo and upgradeToAndCall were having visibility specifiers as `external` and `virtual`, due to this reason derived contract was not able to invoke the function using `super` keyword. I just replaced `external` with `public` so that it can be invoked as well as overridden as expected.

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changelog entry

### CODE REVIEW

**Review by JulissaDantes — CHANGES_REQUESTED** [review]
Please check the other contracts using the `UUPSUpgradeable` contract. All the check must pass, and right now the library its not compiling.

File: .changeset/five-ducks-develop.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (Amxx) [inline .changeset/five-ducks-develop.md:1]: This is not a valid changeset

**Review by Amxx — CHANGES_REQUESTED** [review]

File: .changeset/five-ducks-develop.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-waqas88) [reply .changeset/five-ducks-develop.md:1]: @JulissaDantes and @frangio 

What does line#2 means?

In line#5, just mentioned UUPSUpgradeable.sol whereas I also made changes to UUPSLegacy.sol and UUPSUpgradeableMock.sol

[the remark this answers is not in the document]
Comment (Amxx) [reply .changeset/five-ducks-develop.md:1]: `UUPSLegacy.sol` and `UUPSUpgradeableMock.sol` are mocks. We don't document changes to the mocks.

[the remark this answers is not in the document]
Comment (JulissaDantes) [reply .changeset/five-ducks-develop.md:1]: Remember to remove the changes to mocks in the changeset.

**Review by frangio — APPROVED** [review]
Thanks.

**Pull request merged**